src/OpenLoco/GameCommands/ChangeLandMaterial.cpp

```diff
@@ -0,0 +1,65 @@
+#include "../Map/MapGenerator.h"
+#include "../Map/TileLoop.hpp"
+#include "../Map/TileManager.h"
+#include "../Objects/LandObject.h"
+#include "../Objects/ObjectManager.h"
+#include "GameCommands.h"
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    static loco_global<uint8_t, 0x009C871C> byte_9C871C;
+
+    // 0x00468EDD
+    static void changeLandMaterial(Map::Pos2 pointA, Map::Pos2 pointB, uint8_t landType, uint8_t flags)
+    {
+        GameCommands::setExpenditureType(ExpenditureType::Construction);
+        uint8_t x = (pointA.x + pointB.x) / 2 + 16;
+        uint8_t y = (pointA.y + pointB.y) / 2 + 16;
+        auto height = Map::TileManager::getHeight({ x, y });
+        GameCommands::setPosition({ x, y, height.landHeight });
+
+        Map::TilePosRangeView tileLoop{ { pointA }, { pointB } };
+        for (const auto& tilePos : tileLoop)
+        {
+            auto surface = Map::TileManager::get(tilePos).surface();
+            if (surface != nullptr)
+            {
+                if (flags & Flags::apply)
+                {
+                    surface->setTerrain(landType);
+                    if (!surface->hasHighTypeFlag())
+                    {
+                        surface->setVar6SLR5(0);
+                    }
+                    const auto variation = Map::MapGenerator::getRandomTerrainVariation(*surface);
+                    if (variation.has_value())
+                    {
+                        if (!surface->hasHighTypeFlag())
+                        {
+                            surface->setVariation(variation.value());
```

Comment:
unfortunately we can't use `.value()` on std::optional at the moment due to the old version of apple clang that we compile against.

Reply:
Changed it to *.